src/index.js

```diff
@@ -11,6 +11,12 @@ const flatten = (obj, delimiter) => {
   if (typeof obj !== 'object' || isDate(obj)) return obj
 
   const flat = (original, stack, prev) => {
+    if (!Object.values(original).length) {
+      result[prev] = original
```

Comment:
This should verify if `prev` exist before using it as a key. Currently this results in the following behavior:

```javascript
console.log(flatten({}))
// { undefined: {} }
```

Reply:
Good catch! Just added `prev` to the if logic this should address the issue on hand.